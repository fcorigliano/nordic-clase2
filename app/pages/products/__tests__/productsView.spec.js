const React = require('react');
const ProductsView = require('../view');
const { render, screen } = require('@testing-library/react');
const mockProducts = require('../../../../tests/utils/products.json');

describe('EjercitaciÃ³n Integradora - La View de Products', () => {
    let component;
    const i18n = { gettext: text => text };

    beforeEach(() => {
        component = render(<ProductsView i18n={i18n} products={mockProducts}/>);
    });

    it('1) Renderiza', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();        
    });

    it('2) Renderiza una lista de productos', () => {
        const products = screen.getAllByRole('listitem');
        expect(products[0]).toHaveTextContent(/motorola/i);
        expect(products.length).toBe(10);
    });
});
