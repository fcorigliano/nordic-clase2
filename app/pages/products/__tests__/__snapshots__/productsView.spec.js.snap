// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ejercitaci√≥n Integradora - La View de Products 1) Renderiza 1`] = `
<DocumentFragment>
  <li>
    Motorola
  </li>
  <li>
    Samsung
  </li>
  <li>
    Huawei
  </li>
  <li>
    iPhone
  </li>
  <li>
    Samsung2
  </li>
  <li>
    Nokia
  </li>
  <li>
    LG
  </li>
  <li>
    Sony
  </li>
  <li>
    Motorola2
  </li>
  <li>
    Moto-G
  </li>
</DocumentFragment>
`;
